Profiles/utils.py

```diff
@@ -12,12 +15,282 @@ def wrapped_function(*args, **kwargs):
+        ldap.get_group(group).add_member(account, dn=False)
+
+
+def _ldap_remove_member_from_group(account, group):
+    if _ldap_is_member_of_group(account, group):
+        ldap.get_group(group).del_member(account, dn=False)
+
+@lru_cache(maxsize=1024)
+def _ldap_is_member_of_directorship(account, directorship):
+    directors = ldap.get_directorship_heads(directorship)
+    for director in directors:
+        if director.uid == account.uid:
+            return True
+    return False
+
+
+#Getters 
+
+
+def ldap_get_member(username):
+    return ldap.get_member(username, uid=True)
+
+@lru_cache(maxsize=1024)
+def ldap_get_active_members():
+    return _ldap_get_group_members("active")
+
+@lru_cache(maxsize=1024)
+def ldap_get_intro_members():
+    return _ldap_get_group_members("intromembers")
+
+@lru_cache(maxsize=1024)
+def ldap_get_onfloor_members():
+    return _ldap_get_group_members("onfloor")
+
+@lru_cache(maxsize=1024)
+def ldap_get_current_students():
+    return _ldap_get_group_members("current_student")
+
+@lru_cache(maxsize=1024)
+def ldap_get_all_members():
+    return _ldap_get_group_members("member")
+
+@lru_cache(maxsize=1024)
+def ldap_get_groups(account):
+    group_list = account.get("memberOf")
+    groups = []
+    for group_dn in group_list:
+        groups.append(group_dn.split(",")[0][3:])
+    return groups
+
+@lru_cache(maxsize=1024)
+def ldap_get_eboard():
+    members = _ldap_get_group_members("eboard-chairman") + _ldap_get_group_members("eboard-evaluations") + _ldap_get_group_members("eboard-financial") + _ldap_get_group_members("eboard-history") + _ldap_get_group_members("eboard-imps") + _ldap_get_group_members("eboard-opcomm") + _ldap_get_group_members("eboard-research") + _ldap_get_group_members("eboard-secretary") + _ldap_get_group_members("eboard-social")
+
+    return members
+
+
+
+# Status checkers
+
+def ldap_is_active(account):
+    return _ldap_is_member_of_group(account, 'active')
+
+
+def ldap_is_alumni(account):
+    # If the user is not active, they are an alumni.
+    return not _ldap_is_member_of_group(account, 'active')
+
+
+def ldap_is_eboard(account):
+    return _ldap_is_member_of_group(account, 'eboard')
+
+
+def ldap_is_rtp(account):
+    return _ldap_is_member_of_group(account, 'rtp')
+
+
+def ldap_is_intromember(account):
+    return _ldap_is_member_of_group(account, 'intromembers')
+
+
+def ldap_is_onfloor(account):
+    return _ldap_is_member_of_group(account, 'onfloor')
+
+
+def ldap_is_current_student(account):
+    return _ldap_is_member_of_group(account, 'current_student')
+
+
+# Directorships
+
+def ldap_is_financial_director(account):
+    return _ldap_is_member_of_directorship(account, 'financial')
+
+
+def ldap_is_eval_director(account):
+    return _ldap_is_member_of_directorship(account, 'evaluations')
+
+
+def ldap_is_chairman(account):
+    return _ldap_is_member_of_directorship(account, 'chairman')
+
+
+def ldap_is_history(account):
```

Comment:
This file is getting large, consider breaking out into multiple `.py` files

Reply:
done